gpt4all-chat/database.cpp

```diff
@@ -138,6 +176,42 @@ bool removeChunksByDocumentId(QSqlQuery &q, int document_id)
     return true;
 }
 
+bool selectAllUncompletedChunks(QSqlQuery &q, int folder_id, QList<EmbeddingChunk> &chunks) {
+    if (!q.prepare(SELECT_UNCOMPLETED_CHUNKS_SQL))
+        return false;
+    q.addBindValue(folder_id);
+    if (!q.exec())
+        return false;
+    while (q.next()) {
+        EmbeddingChunk i;
+        i.chunk_id = q.value(0).toInt();
+        i.chunk = q.value(1).toString();
+        i.folder_id = q.value(2).toInt();
+        chunks.append(i);
+    }
+    return true;
+}
+
+bool selectCountChunks(QSqlQuery &q, int folder_id, int &count) {
+    if (!q.prepare(SELECT_COUNT_CHUNKS_SQL))
+        return false;
+    q.addBindValue(folder_id);
+    if (!q.exec())
+        return false;
+    if (q.next())
+        count = q.value(0).toInt();
```

Comment:
I'd prefer if we explicitly set the out parameter on success. Since this query is a `select count(...)`, can we do this?
```suggestion
    Q_ASSERT(q.next());
    count = q.value(0).toInt();
```
Otherwise, maybe this:
```suggestion
    count = q.next() ? q.value(0).toInt() : 0;
```